airflow/executors/worker_executor.py

```diff
@@ -0,0 +1,98 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Airflow executor."""
+
+from typing import Any, List, Optional, Tuple, Union
+
+from kombu import Connection, Exchange, Producer, Queue
+
+from airflow.configuration import conf
+from airflow.exceptions import AirflowException
+from airflow.executors.base_executor import BaseExecutor, CommandType
+from airflow.models.taskinstance import SimpleTaskInstance, TaskInstanceKeyType
+
+task_exchange = Exchange("tasks", type="direct")
+
+
+class AirflowExecutor(BaseExecutor):
+    def __init__(self):
+        super().__init__()
+
+        self.amq_url = conf.get("worker", "broker_url")
+        with Connection(self.amq_url) as conn:
+            self.producer = conn.Producer(serializer="json")
+
+    def start(self) -> None:
+        self.log.debug("Starting Airflow executor %s", self.amq_url)
+
+    def trigger_tasks(self, open_slots: int) -> None:
+        """
+        Overwrite trigger_tasks function from BaseExecutor
+
+        :param open_slots: Number of open slots
+        :return:
+        """
+        # Order tasks by priority
+        sorted_queue = sorted(
+            [(k, v) for k, v in self.queued_tasks.items()],
+            key=lambda x: x[1][1],
+            reverse=True,
+        )
+
+        # Publish tasks
+        for _ in range(min((open_slots, len(self.queued_tasks)))):
+            key, (command, _, queue_name, simple_ti) = sorted_queue.pop(0)
+            queue = Queue(queue_name, task_exchange, routing_key=queue_name)
+            self.producer.publish(
+                {"cmd": command},
+                exchange=task_exchange,
+                routing_key=queue.routing_key,
+                declare=[queue],
+                # retry=True,
+                # retry_policy={
+                #     'interval_start': 0,  # First retry immediately,
+                #     'interval_step': 2,  # then increase by 2s for every retry.
+                #     'interval_max': 3,  # but don't exceed 3s between retries.
+                #     'max_retries': 1,   # give up after 1 tries.
+                # },
+            )
+            self.queued_tasks.pop(key)
+            # TODO: do we need to keep this information in executor?
+            # self.running.add(key)
```

Comment:
It seems that this is only used for printing progress. But I am not convinced that updating progress is something worth to implement  bidirectional communication.